js/functions.js

```diff
@@ -223,21 +248,34 @@ function setupControls()
 	});
 
 	var baseLayers = new Object();
-	baseLayers[translations['mapnik']] = mapnik;
-	baseLayers[translations['mapnikGrayscale']] = mapnikGray;
-	baseLayers[translations['blank']] = blank;
+	baseLayers[_("Mapnik")] = mapnik;
+	baseLayers[_("Mapnik Grayscale")] = mapnikGray;
+	baseLayers[_("No background map")] = blank;
 
 	// hillshading layer
 	var hillshading = new L.TileLayer('http://{s}.tiles.wmflabs.org/hillshading/{z}/{x}/{y}.png',
 	{
-		attribution: translations['hillshadingAttribution'],
+		attribution: _("Hillshading by <a href='http://nasa.gov/'>NASA SRTM</a>"),
 		maxZoom: 17
 	});
 
 	var overlays = new Object();
-	overlays[translations['hillshading']] = hillshading;
-	overlays[translations['railmap']] = railmap;
+	overlays[_("Hillshading")] = hillshading;
+	overlays[_("OpenRailwayMap")] = railmap;
 
 	new L.Control.Scale({metric:true, maxWidth:200}).addTo(map);
 	new L.Control.Layers(baseLayers, overlays).addTo(map);
 }
+
+function _(sourcemsg, n)
+{
+	if (typeof translations[sourcemsg] === 'undefined')
+		return sourcemsg;
+
+	if (n === undefined)
+		return translations[sourcemsg][1];
+
+	var pluralForms = translations[""]["Plural-Forms"].match(/plural=(.*);/)[1];
```

Comment:
This is a bit too simplified to properly handle even western European languages, but it is good enough for German and English. Can always be improved later (preferably by reusing other peoples code).

Reply:
What is the problem with this code? In my opinion it works for all languages we currently have in the repository.